docs/api-docs/analytics/bodl-ga4.md

```diff
@@ -0,0 +1,216 @@
+
+## BODL fetches and stores event data
+
+BODL is a predefined JSON object that extracts data from events. When a merchant enables BODL, BODL object is added to each page of a merchant store. BODL is rendered as a part of the initial DOM load for all pages of all storefronts. 
+
+BODL holds data needed for you to make analytic reports. When a shopper triggers an event, BODL fetches and stores the event and its parameters. For supported events and parameters, see [Events](#events).
+
+Below are examples of BODL after a shopper triggers a start checkout event or a purchase order event:
+
+<!--
+type: tab
+title: Start checkout
+-->
+
+```json title="BODL object after start checkout event" lineNumbers
+{
+    "session": {
+        "first_touch_referral_url": "https://store-{store_hash}.mybigcommerce.com/",
+        "first_touch_request_url": "https://{store_domain}.mybigcommerce.com/",
+        "first_touch_timestamp": "2022-09-22T16:21:15",
+        "id": "123"
+    },
+    "shopper": {
+        "customer_id": null,
+        "email": null,
+        "first_name": null,
+        "last_name": null,
+        "visitor_id": "" 
+    },
+    "data_consent": {
+        "advertising": true,
+        "analytics": true,
+        "functional": true
+    },
+    "events": [
+        "create_checkout_begin"
+    ]
+}
+```
+
+<!--
+type: tab
+title: Purchase order
+-->
+
+```json title="BODL object after order purchase event" lineNumbers
+{
+    "session": {
+        "first_touch_referral_url": "https://store-{store_hash}.mybigcommerce.com/",
+        "first_touch_request_url": "https://{store_domain}.mybigcommerce.com/",
+        "first_touch_timestamp": "2022-09-22T16:21:15",
+        "id": "123"
+    },
+    "shopper": {
+        "customer_id": null,
+        "email": null,
+        "first_name": null,
+        "last_name": null,
+        "visitor_id": "" 
+    },
+    "data_consent": {
+        "advertising": true,
+        "analytics": true,
+        "functional": true
+    },
+    "events": [
+        "create_order_purchased"
+    ]
+}
+```
+
+<!-- type: tab-end -->
+
+<!-- theme: info -->
+> #### Customizing BODL
+> Stencil merchants can turn off the BODL and deploy their own custom version of the BODL. In this scenario, the merchant is choosing a custom route and is expected to ensure that analytics tracking scripts still work.
+
+## Analytic tracking scripts listen for events 
+
+You can pull storefront data from BODL to your solution using analytic storefront tracking scripts. Your analytic scripts will need to listen to the event and take in data that are stored in BODL.  
+
+Below is example code for your analytics script consuming the event. For more info, consult the MDN website doc [Add Event Listeners](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener).
+
+```js title="Add event listener" lineNumbers
+document.addEventListener('checkout_begin', function (bodl_event) {
```

Comment:
@jairo-bc Should we mention we have two strategies here and recommended one?

- subscribe via `bodlEvents` - `window.bodlEvents.checkout.checkoutBegin`
- subscribe via `document` - `document.addEventListener('checkout_begin')`

and what the differences and advantages

Reply:
Also, secure way to subscribe only after the full page loads after all scripts

```
if (window.addEventListener) {
    window.addEventListener("load", functionWithCustomMerchantSubscribers, false)
}
```

Reply:
ping @jairo-bc